File: Task/Symmetric-difference/JavaScript/symmetric-difference-2.js
function Difference(A,B)
{
    var a = A.length, b = B.length, c = 0, C = [];
    for (var i = 0; i < a; i++)
     { var j = 0, k = 0;
       while (j < b && B[j] !== A[i]) j++;
       while (k < c && C[k] !== A[i]) k++;
       if (j == b && k == c) C[c++] = A[i];
     }
    return C;
}

function SymmetricDifference(A,B)
{
    var D1 = Difference(A,B), D2 = Difference(B,A),
        a = D1.length, b = D2.length;
    for (var i = 0; i < b; i++) D1[a++] = D2[i];
    return D1;
}


/* Example
   A = ['John', 'Serena', 'Bob', 'Mary', 'Serena'];
   B = ['Jim', 'Mary', 'John', 'Jim', 'Bob'];

   Difference(A,B);           // 'Serena'
   Difference(B,A);           // 'Jim'
   SymmetricDifference(A,B);  // 'Serena','Jim'
*/

File: Task/Symmetric-difference/OCaml/symmetric-difference-2.ml
# let a = [ "John"; "Bob"; "Mary"; "Serena" ]
  and b = [ "Jim"; "Mary"; "John"; "Bob" ]
  ;;
val a : string list = ["John"; "Bob"; "Mary"; "Serena"]
val b : string list = ["Jim"; "Mary"; "John"; "Bob"]

# a -|- b ;;
- : string list = ["Jim"; "Serena"]

# a -| b ;;
- : string list = ["Serena"]

# b -| a ;;
- : string list = ["Jim"]
